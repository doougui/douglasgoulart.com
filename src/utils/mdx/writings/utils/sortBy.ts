import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
