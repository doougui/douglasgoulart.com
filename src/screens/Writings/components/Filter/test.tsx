import { render, screen } from 'utils/tests';
import userEvent from '@testing-library/user-event';

import { Filter } from '.';

describe('<Filter />', () => {
  it('should render correctly', () => {
    const { container } = render(<Filter />);

    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render search', () => {
    render(<Filter />);

    expect(screen.getByPlaceholderText('Search articles')).toBeInTheDocument();
  });

  it('should render order filter', async () => {
    render(<Filter />);

    await userEvent.selectOptions(screen.getByRole('listbox'), 'Top');

    expect(
      screen.getByRole<HTMLOptionElement>('option', { name: 'New' }).selected,
    ).toBe(false);

    expect(
      screen.getByRole<HTMLOptionElement>('option', { name: 'Top' }).selected,
    ).toBe(true);

    expect(
      screen.getByRole<HTMLOptionElement>('option', { name: 'Hot' }).selected,
    ).toBe(false);
  });
});
