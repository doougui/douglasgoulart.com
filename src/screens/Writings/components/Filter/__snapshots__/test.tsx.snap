// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Filter /> should render correctly 1`] = `
.c4 {
  all: unset;
  width: 100%;
  font-size: 1.6rem;
  background: #2D2D2F;
  padding: 1.2rem 1.5rem;
  border-radius: 0.4rem;
  color: #EDEDEF;
}

.c4::-webkit-input-placeholder {
  color: #888888;
}

.c4::-moz-placeholder {
  color: #888888;
}

.c4:-ms-input-placeholder {
  color: #888888;
}

.c4::placeholder {
  color: #888888;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 2rem;
}

.c1 {
  background: #1C1C1C;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 5px;
  padding: 0 1.5rem;
}

.c2 {
  font-size: 1.4rem;
  line-height: 1.6rem;
  color: #888888;
  white-space: nowrap;
}

.c5 {
  border-top-left-radius: 0;
  border-bottom-left-radius: 0;
}

.c3 {
  background: transparent;
  color: inherit;
  border: none;
  font-weight: 600;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <span
      class="c2"
    >
      Order by:
       
      <select
        aria-label="Select order"
        class="c3"
        data-testid="select-order"
        role="listbox"
      >
        <option>
          New
        </option>
        <option>
          Top
        </option>
        <option>
          Hot
        </option>
      </select>
    </span>
  </div>
  <input
    class="c4 c5"
    placeholder="Search articles"
  />
</div>
`;
